### ISSUE TITLE
Please improve documentation for layouts and workspaces

### ISSUE THREAD

**Issue by randomphrase** [syl20bnr/spacemacs#8587]
As [advised](https://gitter.im/syl20bnr/spacemacs?at=58d830bb4cb8d091732c321f) by @duianto in gitter.im, I am raising this as an issue.

By way of background, let me just say that although I am new to spacemacs, I am not new to emacs. I am familiar with emacs window/buffer/frame management. In adopting spacemacs I would like to learn about the features available - and the documentation provided does not provide a useful introduction, for me at least.

Consider the opening paragraph in section 13:

> Layouts are window configurations with buffer isolation, each layout can define several workspaces (think of them as sub-layouts) sharing the same list of buffers as their parent layout.

Here are some of the questions that occur to me when reading (and re-reading) this sentence.

* What is meant by "window configuration"? I assume this is the same as the emacs definition, but it might be helpful to re-state this here.
* What is meant by "buffer isolation"? This seems to be a spacemacs-specific term.
* What are workspaces, which can be defined by layouts? How do they differ from layouts? The parenthetical comment to think of them as sub-layouts doesn't really help because we are still in the sentence which is defining layouts.
* It is implied that layouts are hierarchical in nature, but what does this mean?

In short I think that a more detailed introduction to this feature would be extremely beneficial. Ideally this introduction would:

* Review any relevant emacs-specific terminology (such as window configuration)
* Define carefully any spacemacs-specific terminology, preferably with illustrations and examples.
* Clarify how the features work together
* Most importantly, provide some common use cases which may aid understanding; why would people want to use these features?

Given that this is an intrinsically visual feature - more screenshots would definitely help also.

Thanks for reading.

**Comment by bmag** [comment]
see also #6200

**Comment by quicknir** [comment]
I'll try to answer these questions, and give my opinion on how clear/unclear they each are:

 - Window configuration is exactly that: a configuration of windows. The number, and position, of windows, on your screen. If you're familiar with emacs, then you know what a window is: something that can show a buffer. I think for emacs users this should be clear.
 - Buffer isolation basically means that when you do `SPC b b`, each layout has its own list of buffers that are separate. It also affects things like searching in all buffers with `SPC s b`. I think the problem here is that to explain this you really need to list concretely the major things affected by it, but it sort of breaks the flow of explanation. Maybe this section can be considered a broad overview and a paragraph below can get into more details.
 - I agree that the sentences for Workspaces is pretty unclear. Basically Workspaces have a window cofiguration, but not their own buffer list. Each layout has one buffer list. Each layout can have one or more workspaces, each of which has the same buffer list, but is a separate window configuration.
 - I think the hierarchy is made pretty clear because it says that Workspaces are sub-layouts.

I generally agree with all of your suggestions for improvement, someone just has to sit down and do the work :-). The emacs terminology should probably have its own section very early on. That way the window and buffer stuff, which is much more basic and widely used, can benefit as well. That section should be called "New to Emacs?" or something similar.

I'd probably kick off by explaining buffer isolation and window configuration *first*, why these might be useful if you're working on different things. That way you can make those two features completely clear before even getting into Workspaces & Layouts. Then explain Workspaces and Layouts. And then, ideally, a video of someone doing a few basic operations.
